src/authenticator.ts

```diff
@@ -89,25 +98,30 @@ export default class Authenticator {
       access_token: newAccessToken,
       token_type: "bearer",
       expires_in: TOKEN_EXPIRY,
-      refresh_token: newRefreshToken,
+      refresh_token: newRefreshToken.value,
     });
   }
 
-  private generateAccessToken(options: AuthenticateOptions): string {
+  generateAccessToken(options: AuthenticateOptions): TokenWithExpire {
     let now = Math.floor(Date.now() / 1000);
+    let expire = now + TOKEN_EXPIRY - TOKEN_LEEWAY;
 
     let claims = {
       app: this.appId,
       iss: this.appKeyId,
       iat: now - TOKEN_LEEWAY,
-      exp: now + TOKEN_EXPIRY - TOKEN_LEEWAY,
+      exp: expire,
       sub: options.userId,
+      ...options.serviceClaims,
     };
 
-    return jwt.sign(claims, this.appKeySecret);
+    return {
+      token: jwt.sign(claims, this.appKeySecret),
+      expire: expire,
```

Comment:
Ah yeah, you're right. The `expires_at` is what's returned to the client. The expiry time returned here is what is used by the feeds server library when it generates a server token.

I guess it would be more consistent to modify this to return an `expiresAt` instead.